src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Semantic/DefaultRazorSemanticTokenInfoServiceTest.cs

```diff
@@ -563,6 +563,36 @@ public async Task GetSemanticTokens_Razor_FunctionsDirectiveAsync()
 
             await AssertSemanticTokensAsync(txt, isRazor: true);
         }
+
+        [Fact]
+        public async Task GetSemanticTokens_Razor_NestedIfDirectives()
+        {
+            var txt = @$"@functions {{
+                private void BidsByShipment(string generatedId, int bids)
+                {{
```

Comment:
Why two `{{`? Is it an escape sequence?

Reply:
Yeah, 2 `{` is the escape sequence for `{`.